service/src/main/kotlin/fi/espoo/evaka/reports/OccupancyReport.kt

```diff
@@ -470,41 +590,41 @@ WHERE sn.placement_id = ANY(:placementIds)
         return assistanceCoefficient * serviceNeedCoefficient
     }
 
-    val placementsAndPlans = (placements + placementPlans).groupBy { it.unitId }
+    val placementsAndPlans = (placements + placementPlans).groupBy { it.groupingId }
 
-    return caretakers
-        .map { (unit, values) ->
+    return caretakerCounts
+        .map { (key, values) ->
             val occupancies = values
-                .associate { (_, date, caretakers) ->
-                    val unitPlacementsOnDate = (placementsAndPlans[unit.unitId] ?: listOf())
-                        .filter { it.period.includes(date) }
-                        .filterNot { absences[it.childId]?.any { absence -> absence.date == date } ?: false }
+                .associate { caretakers ->
+                    val unitPlacementsOnDate = (placementsAndPlans[key.groupingId] ?: listOf())
+                        .filter { it.period.includes(caretakers.date) }
+                        .filterNot { absences[it.childId]?.any { absence -> absence.date == caretakers.date } ?: false }
 
                     val sum = unitPlacementsOnDate
                         .groupBy { it.childId }
-                        .mapNotNull { (_, placements) -> placements.map { getCoefficient(date, it) }.maxOrNull() }
+                        .mapNotNull { (_, placements) ->
```

Comment:
nitpick: `sum` could be renamed to be more specific - now need to read ahead to figure out sum of what.